mm2src/mm2_core/src/mm_ctx.rs

```diff
@@ -324,6 +324,94 @@ impl MmCtx {
     #[cfg(not(target_arch = "wasm32"))]
     pub fn shared_dbdir(&self) -> PathBuf { path_to_dbdir(self.conf["dbdir"].as_str(), self.shared_db_id()) }
 
+    /// Returns the path to the global common directory.
+    ///
+    /// Such directory isn't bound to a specific seed/wallet or address.
+    /// Data that should be stored there is public and shared between all seeds and addresses (e.g. stats, block headers, etc...).
+    #[cfg(all(feature = "new-db-arch", not(target_arch = "wasm32")))]
+    pub fn global_dir(&self) -> Result<PathBuf, String> {
+        let path = path_to_db_root(self.conf["dbdir"].as_str()).join("global");
+        if !path.exists() {
+            std::fs::create_dir_all(&path).map_err(|e| format!("Failed to create global directory: {}", e))?;
```

Comment:
i opted for this io to be sync to avoid the ripple effect it will cause if it's async (it is A LOT esp in address db usage).

The path will probably always exist (for global and wallet dirs) anyways unless this is the first run.
For address dirs, "first runs" happen more frequently, but still, by no means a lot.